discovery-service/src/main/java/org/zowe/apiml/discovery/ApimlInstanceRegistry.java

```diff
@@ -190,4 +201,40 @@ public boolean statusUpdate(String appName, String instanceId, InstanceInfo.Inst
         this.appCntx.publishEvent(new EurekaStatusUpdateEvent(this, appName, instanceId));
         return isUpdated;
     }
+
+    /**
+     * Change the service ID prefix according to the regex before the service registers to Eureka.
+     * @param info the instance info
+     * @param regex the regex
+     * @return instance info with the modified service ID
+     */
+    protected InstanceInfo changeServiceId(final InstanceInfo info, String regex) {
+        String servicePrefix = regex.split(",")[0];
+        String targetValue = regex.split(",")[1];
```

Comment:
only 1 serviceID replacement will be supported?

Reply:
We agreed that for the first PR we will start with this limitation to verify the logic and then we can extend it.

Reply:
Yes, we might enhance it in the future.